src/main/java/com/imsweb/datagenerator/naaccr/NaaccrDataGenerator.java

```diff
@@ -305,6 +316,40 @@ public NaaccrDataGeneratorRule getTumorRule(String ruleId) {
         return tumors;
     }
 
+    /**
+     * Generates a list of values that we want this patient to have.
+     * Specifically the sex of the patient and the sites for the tumors.
+     * <br/><br/>
+     * @param numTumors number of tumors to generate, must be greater than 0
+     */
+    protected Map<String, String> generateInitialPatientContext(int numTumors) {
+
+        // Context contains information that is shared amongst the rules. Contains values we want to use for this patient.
+        Map<String, String> context = new HashMap<>();
+
+        // Pick the patient sex. The sex of the patient will influence the tumor sites.
+        context.put("sex", DistributionUtils.getSex());
+
+        // Pick the sites we want for the tumors. The Tumor sites can influence the patient's age.
+        context.put("currentTumor", "");
+        context.put("totalTumorCount", String.valueOf(numTumors));
+        for (int i = 0; i < numTumors; i++) {
+            SiteFrequencyDto dto = DistributionUtils.getSite(context.get("sex"));
```

Comment:
If you switch to a map of String/Object for the context, you should just put the SiteFrequencyDto in the context. In fact, I wouldn't try to use keys based on the tumor index; that's a bit hackish. I would add a map into the context that under the key "siteFrequencies" or something similar; that inner map would be Integer/SiteFrequencyDto, the keys would be the tumor index and values the frequencies. Another solution would be to use an ArrayList and use the tumor index as the array index, but I think the map is more elegant (less hackish).

Reply:
Fixed.